File: DevExpress-Demo/VB/PropertyGridDemo.Wpf/ViewModels/TabbedViewModule/Model/Metadata/Effects.vb
Imports DevExpress.Mvvm.DataAnnotations

Namespace PropertyGridDemo.Metadata

    Public Module MirrorOptionsMetadata

        Public Sub BuildMetadata(ByVal builder As MetadataBuilder(Of MirrorOptions))
            builder.Property(Function(x) x.Show).PropertyGridEditor("MirrorOptions.Show")
            builder.Property(Function(x) x.MirrorLength).PropertyGridEditor("MirrorOptions.MirrorLength").LocatedAt(0)
        End Sub
    End Module

    Public Module LabelOptionsMetadata

        Public Sub BuildMetadata(ByVal builder As MetadataBuilder(Of LabelOptions))
            builder.Property(Function(x) x.ShowLabel).PropertyGridEditor("LabelOptions.ShowLabel").LocatedAt(0)
        End Sub
    End Module

    Public Module VisibleLabelOptionsMetadata

        Public Sub BuildMetadata(ByVal builder As MetadataBuilder(Of VisibleLabelOptions))
            builder.Property(Function(x) x.Position).PropertyGridEditor("VisibleLabelOptions.Position")
        End Sub
    End Module
End Namespace

File: DevExpress-Demo/CS/PropertyGridDemo.Wpf/ViewModels/TabbedViewModule/Model/Metadata/Effects.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PropertyGridDemo.Metadata {
    public static class MirrorOptionsMetadata {
        public static void BuildMetadata(MetadataBuilder<MirrorOptions> builder) {
            builder.Property(x => x.Show).PropertyGridEditor("MirrorOptions.Show");
            builder.Property(x => x.MirrorLength).PropertyGridEditor("MirrorOptions.MirrorLength").LocatedAt(0);
        }
    }
    public static class LabelOptionsMetadata {
        public static void BuildMetadata(MetadataBuilder<LabelOptions> builder) {
            builder.Property(x => x.ShowLabel).PropertyGridEditor("LabelOptions.ShowLabel").LocatedAt(0);
        }
    }
    public static class VisibleLabelOptionsMetadata {
        public static void BuildMetadata(MetadataBuilder<VisibleLabelOptions> builder) {
            builder.Property(x => x.Position).PropertyGridEditor("VisibleLabelOptions.Position");
        }
    }
}
